### PR TITLE
feat: implement simplified unbonding and rewards queries

### PR DESCRIPTION
**Pull request by emidev98**
As per the following issue https://github.com/terra-money/alliance/issues/279 this PR implements a simplified way to query the unbondings and redelegations. It also includes tests and fixes for the other queries that does not return the right values.

These features were requested by @arthuryeti to simplify building frontends

### CODE REVIEW

**Review by javiersuweijie — CHANGES_REQUESTED** [review]
LGTM, but missing tests

File: proto/alliance/alliance/query.proto [hunk 1]
```diff
@@ -249,7 +249,6 @@ message QueryAllianceUnbondingsByDelegatorResponse {
   option (gogoproto.goproto_getters) = false;
 
   repeated UnbondingDelegation unbondings = 1 [(gogoproto.nullable) = false];
-  cosmos.base.query.v1beta1.PageResponse pagination = 2;
```
Comment (javiersuweijie) [inline proto/alliance/alliance/query.proto]: why we remove this?

**Review by javiersuweijie — COMMENTED** [review]
Still missing the tests for the new API

File: proto/alliance/alliance/query.proto (hunk 1 above)
Comment (emidev98) [reply proto/alliance/alliance/query.proto]: Because we don't have a way to paginate that specific request and the amount of items to be recovered from the API is number_of_alliances * 7 (maximum undelegations)

**Comment by emidev98** [comment]
@javiersuweijie the tests for the new API are added here:

- AllianceRedelegationsByDelegator: https://github.com/terra-money/alliance/pull/313/files#diff-236f586ff7f57b977fcae2389f81edf3a4c75567fc9f2c8e0bb2c01c0f66e7c7R337-R354
- AllianceUnbondingsByDelegator: https://github.com/terra-money/alliance/pull/313/files#diff-236f586ff7f57b977fcae2389f81edf3a4c75567fc9f2c8e0bb2c01c0f66e7c7R672-R692

If you want I can add more tests or I can try covering edge cases

**Review by javiersuweijie — APPROVED** [review]
